programs/openbook-v2/src/instructions/place_order.rs

```diff
@@ -73,28 +69,69 @@ pub fn place_order(ctx: Context<PlaceOrder>, order: Order, limit: u8) -> Result<
     let (to_vault, deposit_amount) = match side {
         Side::Bid => {
             let free_assets_native = position.quote_free_native;
-            let max_native_including_fees = I80F48::from_num(max_quote_lots_including_fees)
-                * I80F48::from_num(market.quote_lot_size);
 
-            let min_qua = cmp::min(max_native_including_fees, free_assets_native);
-            position.quote_free_native -= min_qua;
+            let max_native_including_fees: I80F48 = match order.params {
+                OrderParams::Market | OrderParams::ImmediateOrCancel { .. } => {
+                    total_quote_taken_native
+                }
+                OrderParams::Fixed {
+                    price_lots: _,
+                    order_type,
+                } => {
+                    // For PostOnly If existing orders can match with this order, do nothing
+                    if order_type == PostOrderType::PostOnly && order_id.is_none() {
+                        I80F48::ZERO
+                    } else {
+                        I80F48::from_num(max_quote_lots_including_fees)
```

Comment:
this looks like is taking more amount than required. the amount that's placed in the book is `let book_base_quantity = remaining_base_lots.min(remaining_quote_lots / price_lots);` not the max_lots passed as input args